mmv1/third_party/terraform/services/container/resource_container_cluster.go.tmpl

```diff
@@ -3122,19 +3240,19 @@ func resourceContainerClusterUpdate(d *schema.ResourceData, meta interface{}) er
 	// The ClusterUpdate object that we use for most of these updates only allows updating one field at a time,
 	// so we have to make separate calls for each field that we want to update. The order here is fairly arbitrary-
 	// if the order of updating fields does matter, it is called out explicitly.
-	if d.HasChange("master_authorized_networks_config") {
-		c := d.Get("master_authorized_networks_config")
+
+	if d.HasChange("control_plane_endpoints_config") || d.HasChange("master_authorized_networks_config") {
 		req := &container.UpdateClusterRequest{
 			Update: &container.ClusterUpdate{
-				DesiredMasterAuthorizedNetworksConfig: expandMasterAuthorizedNetworksConfig(c, d),
+				DesiredControlPlaneEndpointsConfig: expandControlPlaneEndpointsConfig(d),
```

Comment:
Will this cause an issue for a config with both `master_authorized_networks_config` and `private_cluster_config` set? Do any test cases do that already so we can be sure there aren't regressions?

Reply:
Thanks! Yeah, I was concerned about potentially sending a partial DesiredControlPlaneEndpointsConfig and overwriting previously set values for the other field.